src/wayland-eglsurface.c

```diff
@@ -1199,14 +1201,20 @@ wlEglSurfaceCheckReleasePoints(WlEglDisplay *display, WlEglSurface *surface)
      * Note that there are some bugs with older kernels where this may not
      * signal correctly.
      */
-    if (drmSyncobjTimelineWait(display->drmFd, syncobjs, syncPoints,
-                               surface->ctx.numStreamImages, timeout,
-                               DRM_SYNCOBJ_WAIT_FLAGS_WAIT_AVAILABLE,
-                               &firstSignaled) != 0) {
-        goto end;
+    {
```

Comment:
What's the purpose of this bracketed block now that the if condition is gone? Is it just so `syncobjWaitRes` has a bounded lifetime? If so I think we should get rid of the extra brackets and declare `syncobjWaitRes` above with the rest of the local variables.

Reply:
I believe it is a good practice to keep the scope of the variables minimal. The rest of the code in this function does not need to know the returned value.  But I do not mind removing this scope and moving the declaration of the variable if you prefer that way.

Reply:
Limiting variable scope is nice when you can do it, but not at the readability cost of disrupting the block structure.

Reply:
Stylistically this isn't something we really seem to do, but I think if you want to attain the same scope reduction we could create a small helper function to wrap `drmSyncobjTimelineWait`. I think that could look nicer than this block scoping

Reply:
In this case, you could also remove `syncobjWaitRes` entirely and just check `errno` in your assert.

Reply:
Checking errno makes sense instead of using a separate variable, which helps to clean this up. Thank you for all the suggestions.